src/auto-evo/steps/ForceExtinction.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace AutoEvo
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Godot;
+
+    /// <summary>
+    ///   Forces extinction of worse-faring species to limit the number of steps in next auto-evo run
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     This was a hotfix to the 0.5.6 release.
+    ///     TODO: Inquire more refined solution, such as random extinction with bias for smaller populations.
+    ///   </para>
+    /// </remarks>
+    public class ForceExtinction : IRunStep
+    {
+        private readonly List<Patch> patches;
+        private readonly AutoEvoConfiguration configuration;
+
+        public ForceExtinction(List<Patch> patches, AutoEvoConfiguration configuration)
+        {
+            this.patches = patches;
+            this.configuration = configuration;
+        }
+
+        public bool CanRunConcurrently => false;
+
+        public int TotalSteps => 1;
+
+        public bool RunStep(RunResults results)
+        {
+            var populationsByPatch = results.GetPopulationsByPatch(!configuration.ProtectMigrationsFromSpeciesCap,
+                !configuration.AllowNoMutation, !configuration.ProtectNewCellsFromSpeciesCap);
+
+            foreach (Patch patch in patches)
+            {
+                var speciesInPatch = populationsByPatch[patch];
+
+                // This does not take player into account as the species can never go extinct this way.
+                var protectedSpeciesCount = 0;
+
+                if (configuration.ProtectNewCellsFromSpeciesCap)
+                    protectedSpeciesCount += results.GetNewSpeciesResults(patch).Count;
+                if (configuration.ProtectMigrationsFromSpeciesCap)
+                    protectedSpeciesCount += results.GetMigrationsTo(patch).Count;
+
+                // Only bother if we're above the limit
+                if (speciesInPatch.Count + protectedSpeciesCount <= configuration.MaximumSpeciesInPatch)
+                    continue;
+
+                GD.Print("Running extinction step in patch ", patch.Name, ". ",
+                    "Total count:", speciesInPatch.Count + protectedSpeciesCount);
+
+                // Sort the species in the patch, unless protected species fill up all the place already...
+                var orderedSpeciesInPatch = (configuration.MaximumSpeciesInPatch > protectedSpeciesCount) ?
```

Comment:
Is this ternary here a "useless" optimization. I'm very certain that a OrderBy with less than 100 list items cannot be slow.

Reply:
Does it hurt that much to keep it though? Not to mention that there is no theoretical limit on the number of species in the patch: if we find a way to have a fast auto-evo even with 100 species, then this part would be of interest.

Reply:
Sorting a list with less than 10000 items should be very fast anyway. I think it'll be a big understandability and readability win to not use the ternary operator here. I had to look at this for like 30 seconds to figure this out, whereas without the ternary understanding would be as easy as reading the line.